dataservice/api/common/schemas.py

```diff
@@ -156,6 +154,42 @@ class PaginatedSchema(Schema):
     return PaginatedSchema
 
 
+def filter_schema_factory(model_filter_schema_cls):
+    """
+    Create an instance of a model's filter schema
+
+    Remove schema attributes that are not applicable to filters (_links)
+    Allow partially populated schema
+    Validate with strict=True - reuse model schema's validators
+    """
+    # Exclude fields
+    exclude = ('_links', )
+    if hasattr(model_filter_schema_cls.Meta, 'exclude'):
+        exclude += model_filter_schema_cls.Meta.exclude
+
+    return model_filter_schema_cls(strict=True,
+                                   partial=True,
+                                   exclude=exclude)
+
+
+class FilterSchemaMixin(ma.ModelSchema):
```

Comment:
The base Marshmallow `Schema` may be a better parent class here?

Reply:
Hmm yea probably...I can't remember why I decided to use ModelSchema anymore. I'll change it to `Schema`